### PR TITLE
Maintenance fixes

### PR DESCRIPTION
**Pull request by espkk**
A few general bugfixes + a lot of SCA fixes

### CODE REVIEW

File: src/libs/geometry/src/geometry.cpp [hunk 1]
```diff
@@ -229,7 +229,7 @@ void *GEOM_SERVICE_R::malloc(long bytes)
 
 void GEOM_SERVICE_R::free(void *ptr)
 {
-    delete ptr;
+    delete[] static_cast<char*>(ptr);
```
Comment (WillTer) [inline src/libs/geometry/src/geometry.cpp:232]: What purpose of casting before deletion?

File: src/libs/island/src/ISLAND.cpp [hunk 2]
```diff
@@ -775,7 +775,7 @@ bool ISLAND::Mount(const std::string_view &fname, const std::string_view &fdir,
     if (pNode)
         seabed_id = pNode->Unlink2Model();
     else
-        core.Trace("Island: island %s has no sea bed, check me!", fname);
+        core.Trace("Island: island %s has no sea bed, check me!", std::string(fname).c_str());
```
Comment (WillTer) [inline src/libs/island/src/ISLAND.cpp:778]: Why not `fname.data()`?

File: src/libs/xinterface/src/QuestFileReader/QuestFileReader.cpp [hunk 3]
```diff
@@ -265,15 +265,15 @@ void QuestFileReader::SetQuestTextFileName(const std::string_view &fileName)
     auto fileS = fio->_CreateFile(fileName.data(), std::ios::binary | std::ios::in);
     if (!fileS.is_open())
     {
-        core.Trace("WARNING! Can`t open quest log file %s", fileName);
+        core.Trace("WARNING! Can`t open quest log file %s", std::string(fileName).c_str());
```
Comment (WillTer) [inline src/libs/xinterface/src/QuestFileReader/QuestFileReader.cpp:268]: `fileName.data()`?

File: src/libs/geometry/src/geometry.cpp (hunk 1 above)
Comment (espkk) [reply src/libs/geometry/src/geometry.cpp:232]: this is just not allowed by standard: https://eel.is/c++draft/expr.delete#footnote-69
practically, usually there is no difference, nevertheless it is still UB and treated as a misusage by most modern compilers and SCA tools

File: src/libs/island/src/ISLAND.cpp (hunk 2 above)
Comment (espkk) [reply src/libs/island/src/ISLAND.cpp:778]: because std::string_view has no guarantee that the data is null-terminated. https://eel.is/c++draft/string.view#access-14
actually, that's the key feature of string_view

File: src/libs/xinterface/src/QuestFileReader/QuestFileReader.cpp (hunk 3 above)
Comment (espkk) [reply src/libs/xinterface/src/QuestFileReader/QuestFileReader.cpp:268]: see above

File: src/libs/island/src/ISLAND.cpp (hunk 2 above)
Comment (espkk) [reply src/libs/island/src/ISLAND.cpp:778]: to the point, ideally we should switch to spdlog/fmt formatting completely; that would also solve such kind of issues. recent spdlog update targets fmt 8 support that I believe can help a lot with logging refactoring since it brings stable consteval formatting lexic parser

File: src/apps/ENGINE/src/compiler.cpp [hunk 4]
```diff
@@ -5206,7 +5206,7 @@ bool COMPILER::BC_Execute(uint32_t function_code, DATA *&pVReturnResult, const c
                     return false;
                 if (pVDst->GetType() != VAR_REFERENCE)
                 {
-                    SetError("'%s' isnt reference", real_var->name.c_str());
+                    SetError("Local variable is not reference");
```
Comment (np-c0mp1ete) [inline src/apps/ENGINE/src/compiler.cpp:5203]: why did you remove the var name?

File: src/apps/ENGINE/src/s_dbg_watcherlist.cpp [hunk 5]
```diff
@@ -18,25 +16,22 @@ WATCHER_LIST::WATCHER_LIST(HWND hwnd, HINSTANCE hinst)
...
+        long n = 0;
         sprintf_s(buffer, "E%d", n);
         if (ini->ReadString(nullptr, buffer, buffer, sizeof(buffer), ""))
         {
-            if (n < 11) //~!~
-            {
-                SetItemText(n, 0, buffer);
-                SetItemText(n, 1, CDebug.ProcessExpression(buffer));
-            }
+            SetItemText(n, 0, buffer);
```
Comment (np-c0mp1ete) [inline src/apps/ENGINE/src/s_dbg_watcherlist.cpp:33]: what was the reason for the n < 11 check?

File: src/libs/lighter/src/LGeometry.h [hunk 6]
```diff
@@ -18,12 +18,14 @@ class LGeometry
...
         Object()
+            : model(0)
         {
             name = nullptr;
             nameReal = nullptr;
             m = nullptr;
             lBufSize = 0;
-        };
+        }
+        ;
```
Comment (np-c0mp1ete) [inline src/libs/lighter/src/LGeometry.h:28]: is it on purpose?

**Review by Hammie — DISMISSED** [review]
A bunch of TODO's were added, so it seems we have more to do ;)

I do like seeing a lot less uninitialized variables though :+1:

I also tested it with CoAS and everything seemed to be working.

**Comment by Hammie** [comment]
@np-c0mp1ete Do you have anything to add, or can we go ahead and merge it?

File: src/apps/ENGINE/src/s_stack.cpp [hunk 7]
```diff
@@ -29,10 +29,6 @@ void S_STACK::Release()
 
 DATA *S_STACK::Push(DATA *pdataclass)
 {
-    if (Data_num > 1000)
```
Comment (np-c0mp1ete) [inline src/apps/ENGINE/src/s_stack.cpp:32]: what was the purpose of this check?

File: src/libs/sailors/src/Sailors.cpp [hunk 8]
```diff
@@ -34,7 +35,8 @@ ShipMan::ShipMan()
     jumpSpeedX = jumpSpeedY = 0;
 
     // UN//GUARD_SAILORS
-};
+}
+;
```
Comment (np-c0mp1ete) [inline src/libs/sailors/src/Sailors.cpp:39]: maybe just remove the semicolon?

File: src/libs/sailors/src/Sailors.cpp [hunk 9]
```diff
@@ -1025,7 +1028,8 @@ Sailors::Sailors()
     IsOnDeck = 0;
 
     // UN//GUARD_SAILORS
-};
+}
+;
```
Comment (np-c0mp1ete) [inline src/libs/sailors/src/Sailors.cpp:1032]: remove?

File: src/libs/sailors/src/Sailors.h [hunk 10]
```diff
@@ -50,10 +50,12 @@ struct ShipState
     SEA_BASE *sea;
 
     ShipState()
+        : sea(nullptr)
     {
         mode = SHIP_SAIL;
         dead = false;
-    };
+    }
+    ;
```
Comment (np-c0mp1ete) [inline src/libs/sailors/src/Sailors.h:58]: remove?

File: src/libs/sea_ai/src/AIShip.h [hunk 11]
```diff
@@ -177,12 +177,14 @@ class AIShip : public VAI_INNEROBJ
...
-        AI_POINT(){};
+        AI_POINT() = default;
 
         AI_POINT(VAI_INNEROBJ *_pObj)
+            : fAngle(0)
         {
             pObj = _pObj;
-        };
+        }
+        ;
```
Comment (np-c0mp1ete) [inline src/libs/sea_ai/src/AIShip.h:187]: remove?

File: src/libs/sea_cameras/src/Common_Camera.h [hunk 12]
```diff
@@ -102,11 +102,13 @@ class COMMON_CAMERA : public Entity
...
 
     COMMON_CAMERA()
+        : eidObject(0), pAIObj(nullptr), pACharacter(nullptr)
     {
         bOn = false;
         bActive = false;
         fPerspective = 1.285f;
-    };
+    }
+    ;
```
Comment (np-c0mp1ete) [inline src/libs/sea_cameras/src/Common_Camera.h:111]: remove?

**Review by np-c0mp1ete — COMMENTED** [review]
Thanks for the PR! I added some minor comments/questions, but in general the code looks fine, feel free to merge. I'm not necessarily insisting on addressing my comments.

File: src/apps/ENGINE/src/compiler.cpp (hunk 4 above)
Comment (espkk) [reply src/apps/ENGINE/src/compiler.cpp:5203]: We cannot get its name from here... calling this code will lead to a crash

File: src/apps/ENGINE/src/s_dbg_watcherlist.cpp (hunk 5 above)
Comment (espkk) [reply src/apps/ENGINE/src/s_dbg_watcherlist.cpp:33]: it's a copy-paste from the loop below. it makes no sense here and looks weird here

File: src/libs/lighter/src/LGeometry.h (hunk 6 above)
Comment (espkk) [reply src/libs/lighter/src/LGeometry.h:28]: no. this is auto-formatting (maybe from R#). the semicolon is excessive so I believe the formatter inserted a newline after a finished token

File: src/apps/ENGINE/src/s_stack.cpp (hunk 7 above)
Comment (espkk) [reply src/apps/ENGINE/src/s_stack.cpp:32]: this is typical for engine developers to check for abnormal behavior. it's likely been used during development to detect recursiveness, etc.

**Comment by espkk** [comment]
These newlines before surplus semicolons are the result of R# formatting around processed core areas... I'll check if there is a way to auto-remove all of them

**Comment by espkk** [comment]
I'll remove redundant semicolons with the next global reformatting
